### ISSUE TITLE
duplicate_enum_cases displays error when the same name cases belong to different OSs

### ISSUE THREAD

**Issue by Cyberbeni** [realm/SwiftLint#3558]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

This shouldn't produce an error for duplicate_enum_cases:

```swift 
enum Target {
#if os(iOS)
  case file
#else
  case file(URL)
#endif
}
```

##### Complete output when running SwiftLint, including the stack trace and command used

```console
> swiftlint target.swift
Linting Swift files at paths target.swift
Linting 'target.swift' (1/1)
/Users/benedek.kozma/dev/test/target.swift:3:8: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
/Users/benedek.kozma/dev/test/target.swift:5:8: error: Duplicate Enum Cases Violation: Enum can't contain multiple cases with the same name. (duplicate_enum_cases)
Done linting! Found 2 violations, 2 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
    * 0.43.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
    * Homebrew ARM64
* Paste your configuration file:
    * No config

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
    * 12.4
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.
